study1/week1/Merge_Nodes_in_Between_Zeros.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Definition for singly-linked list.
+ * public class ListNode {
+ *     int val;
+ *     ListNode next;
+ *     ListNode() {}
+ *     ListNode(int val) { this.val = val; }
+ *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
+ * }
+ */
+class Solution {
+    public ListNode mergeNodes(ListNode head) {
+        ListNode dummyHead = new ListNode(0);
```

Comment:
Using `dummyHead`, `newTemp` and `temp` as variable names are confusing to me as it's similar and the reader need to actually read through the code to understand the intent.

Using `current` or `iter` as a name instead of `temp` sounds better as you're iterating over the linked list, and it indicates that this is the current node.

The `dummyHead` looks like it should be a `mergeRoot` as it's the root of the linked list of the newly merged linked list.

for `newTemp` maybe consider using `mergedCurrent` `mergedIter` or something like that should be better.